_posts/2016-02-16-ten-steps-to-evaluate-a-rails-project.markdown

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title: "10 Steps to Evaluate a Rails Project"
+date: 2016-02-16 13:55:00
+categories: ["rails","maintenance"]
+author: "etagwerker"
+---
+
+It will come a time when you will have to decide whether to maintain a [Rails](http://rubyonrails.org) project or not.
+
+If you want to seriously consider it, you should evaluate all these steps.
+
+## 1. Setup the development environment
+
+Git clone the repository and try to start the server. Is the `README` clear enough? Can you follow the steps in the file and easily get started?
+
+A lot of projects will have a `README` that is out of date and/or instructions that don't work right off the bat.
+
+Most of the projects will define guidelines like these:
+
+* Configure your `config/database.yml`
+* Configure your `.env` file
+* Setup the database `rake db:create db:migrate db:seed`
+* Start the server `rails server`
+
+The best projects will have a one-liner that will setup the entire environment for you.
+
+For example:
+
+```bash
+cd path/to/project
+./bin/setup
+rails server
+```
+
+## 2. Run the tests
+
+Once you have setup your development environment, try to run the tests. For example:
+
+```bash
+rake
+```
+
+Any decent [Rails](http://rubyonrails.org) project will have a passing build, even if it's not very thorough.
+
+Hopefully, the build won’t take more than 10 minutes to finish. Some builds will take a long time, especially if they include integration tests (e.g. [Selenium](http://www.seleniumhq.org))
+
+Did the build pass? Did it fail? How many failures did you get?
+
+In the best case scenario, the tests will pass.
+
+## 3. Review schema.rb
+
+A good way to see the complexity of an application is to see the amount of tables in the database. Does it have more than 20 tables? More than 100? Can you easily draw an [ERD](https://en.wikipedia.org/wiki/Entity–relationship_model) of the core models?
+
+I like doing the exercise to draw the core business tables and the relationships between them. This usually helps me understand the business behind the project.
```

Comment:
The first phrase is kinda confusing, maybe "of drawing..." instead of "to draw..."? Not sure how it can be rephrased.